torch/onnx/symbolic_opset13.py

```diff
@@ -109,6 +109,136 @@ def unsafe_split_with_sizes(g, self, split_sizes, dim, _outputs=None):
     return split_with_sizes(g, self, split_sizes, dim, _outputs)
 
 
+@symbolic_helper.parse_args("v", "v", "i", "i")
+def tensor_split(g, self, indices_or_sections, dim, _outputs=None):
+    axis = g.op("Constant", value_t=torch.tensor(dim, dtype=torch.long))
+    axis = opset11.unsqueeze(g, axis, 0)
+
+    def static_tensor_split(self, indices):
+        start = g.op("Constant", value_t=torch.tensor([0], dtype=torch.long))
+        res = []
+        for i in range(_outputs - 1):
+            end = g.op("Gather", indices_or_sections, i, axis)
+            res.append(g.op("Slice", self, start, end, axis))
+            start = end
+
+        end = symbolic_helper._size_helper(g, self, axis)
+        res.append(g.op("Slice", self, start, end, axis))
+        return res
+
+    if not symbolic_helper._is_split_static(indices_or_sections, _outputs):
+        is_scalar = symbolic_helper._get_tensor_rank(indices_or_sections) == 0
+
+        if (
+            symbolic_helper._is_tensor(indices_or_sections) and not is_scalar
+        ):  # 1-d tensor
+            # loop conditions
+            const_1 = g.op("Constant", value_t=torch.tensor(1, dtype=torch.long))
+            loop_len = symbolic_helper._size_helper(
+                g, indices_or_sections, g.op("Constant", value_t=torch.tensor(0))
+            )
+            loop_len = opset11.unsqueeze(g, loop_len, 0)
+            loop_condition = g.op("Cast", const_1, to_i=9)
+
+            padding_0 = g.op("Constant", value_t=torch.tensor([0], dtype=torch.long))
+            indices_or_sections = g.op(
+                "Concat", padding_0, indices_or_sections, axis_i=0
+            )
+
+            # Create an empty sequence to store final expansions
+            final_splits = g.op("SequenceEmpty")
+            loop = g.op("Loop", loop_len, loop_condition, final_splits)
+
+            # Loop inputs
+            loop_block = utils._add_block(loop.node())
+            block_input_iter = utils._add_input_to_block(loop_block)
+            cond = utils._add_input_to_block(loop_block)
+            final_splits = utils._add_input_to_block(loop_block)
+
+            start = loop_block.op(
+                "Gather", indices_or_sections, block_input_iter, axis_i=0
+            )
+            end = loop_block.op(
+                "Gather",
+                indices_or_sections,
+                loop_block.op("Add", block_input_iter, const_1),
+                axis_i=0,
+            )
+
+            slice = loop_block.op("Slice", self, start, end, axis)
+            final_splits = loop_block.op("SequenceInsert", final_splits, slice)
+
+            # Loop outputs
+            cond_out = loop_block.op("Cast", loop_condition, to_i=9)
```

Comment:
why is this cast needed? loop_condition was already cast to bool on line 141

Reply:
No need indeed. I changed it to `Identity`.